engine/worker/internal/action/builtin_artifact_download.go

```diff
@@ -77,7 +78,12 @@ func RunArtifactDownload(ctx context.Context, wk workerruntime.Runtime, a sdk.Ac
 		return res, fmt.Errorf("cds.run.number variable is not valid. aborting")
 	}
 
-	reg, err := regexp.Compile(pattern)
+	interpolatedPattern, err := interpolate.Do(pattern, sdk.ParametersToMap(wk.Parameters()))
```

Comment:
action parameter should be interpolated before. i think this it not the right fix